lib/ds/array.flow

```diff
@@ -611,11 +611,21 @@ split(a : [?], f : (?) -> bool) {
 }
 
 splitByNumber(a : [?], n : int) -> [[?]]{
-	l = length(a);
-	if (l <= n || n <= 0) {
+	if (length(a) <= n || n <= 0) {
 		[a];
 	} else {
-		concat([subrange(a, 0, n)], splitByNumber(subrange(a, n, l-n), n))
+		// Use tail recursion to avoid out of memory error (e.g. with 2M elements in a splitted by 1k)
+		splitByNumber1(a, 0, n, []);
+	}
+}
+
+splitByNumber1(a : [?], i : int, n : int, acc : [[?]]) -> [[?]]{
+	if (length(a) <= n || n <= 0) { 
+		concat(acc, [a]);
+	} else if (length(a) <= i) {
+		acc;
+	} else {
+		splitByNumber1(a, i + n, n, concat(acc, [subrange(a, i, n)]))
```

Comment:
`concat(acc, [subrange(a, i, n)])` can be replaced by `arrayPush(acc, subrange(a, i, n))`